From caadaed6db01545d62bb10d8199177f04a446010 Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Mon, 16 Jun 2014 12:55:57 -0400
Subject: [s390] sclp: replace uninitialized early_event_mask_sccb variable with sccb_early

Message-id: <1402923362-2331-7-git-send-email-brueckner@redhat.com>
Patchwork-id: 84032
O-Subject: [RHEL7.1 PATCH 06/11] [s390] s390/sclp: replace uninitialized early_event_mask_sccb variable with sccb_early
Bugzilla: 1107730
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit 9e3ea19e35635ecd8373fc04f5dfb072be5f6d2c
Author: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
Date:   Fri Nov 29 17:29:20 2013 +0100

    s390/sclp: replace uninitialized early_event_mask_sccb variable with sccb_early

    Commit "s390/sclp: Consolidate early sclp init calls to sclp_early_detect()"
    (7b50da53f6ad2048241bef232bfc22a132a40283) replaced the sclp_event_mask_early()
    with sclp_set_event_mask().  The early_event_mask_sccb variable is no longer
    initialized but is still used in sclp_has_linemode() and sclp_has_vt220().

    Replace early_event_mask_sccb with the sccb_early variable in both
    functions.

    Signed-off-by: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
    Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>

Bugzilla
--------
BZ 1107730
https://bugzilla.redhat.com/show_bug.cgi?id=1107730

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=9e3ea19e35635ecd8373fc04f5dfb072be5f6d2c

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7572070

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/s390/char/sclp_early.c b/drivers/s390/char/sclp_early.c
index f7aa080..1465e95 100644
--- a/drivers/s390/char/sclp_early.c
+++ b/drivers/s390/char/sclp_early.c
@@ -35,7 +35,6 @@ struct read_info_sccb {
  u8 _reserved5[4096 - 112]; /* 112-4095 */
 } __packed __aligned(PAGE_SIZE);
 
-static __initdata struct init_sccb early_event_mask_sccb __aligned(PAGE_SIZE);
 static __initdata struct read_info_sccb early_read_info_sccb;
 static __initdata char sccb_early[PAGE_SIZE] __aligned(PAGE_SIZE);
 static unsigned long sclp_hsa_size;
@@ -113,7 +112,7 @@ static void __init sclp_facilities_detect(void)
 
 bool __init sclp_has_linemode(void)
 {
- struct init_sccb *sccb = &early_event_mask_sccb;
+ struct init_sccb *sccb = (void *) &sccb_early;
 
  if (sccb->header.response_code != 0x20)
   return 0;
@@ -126,7 +125,7 @@ bool __init sclp_has_linemode(void)
 
 bool __init sclp_has_vt220(void)
 {
- struct init_sccb *sccb = &early_event_mask_sccb;
+ struct init_sccb *sccb = (void *) &sccb_early;
 
  if (sccb->header.response_code != 0x20)
   return 0;
-- 
1.7.1